lib/Doctrine/DBAL/Cache/QueryCacheProfile.php

```diff
@@ -99,10 +101,14 @@ public function getCacheKey()
      */
     public function generateCacheKeys($query, $params, $types, array $connectionParams = [])
     {
+        $hashedParams = array_map(function ($param) {
+            return crypt($param, md5($param));
```

Comment:
This approach assumes that all `$param`s are scalar which may be true for now but not necessarily true in the future. Why not just hash the serialized representation once instead of hashing parameters individually?

Reply:
I'm not a security expert but `md5($param)` is not random (the salt should be random). Should something like `random_bytes()` be used instead, or can we just get away with something like `md5()` or `sha1()` for this purpose?

Reply:
First one seems legit, since I don't know where is it really called I didn't want to change too much stuff. If it's ok I'll change it.
Salt is not random because with same strings we want same outputs!